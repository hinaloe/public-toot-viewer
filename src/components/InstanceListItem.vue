<template>
  <a
    class="dropdown-item"
    href="#"
    @click.prevent="$emit('selected', instance.name)"
  >
    <div class="instance-name">{{ instance.name }}</div>
    <div>
      <small>
        <b
          v-if="!instance.up"
          class="text-danger"
        >DOWN </b>
        {{ instance.users }} users / connections: {{ instance.connections }}
      </small>
    </div>
  </a>
</template>
<script setup lang="ts">
type InstanceInfo = {
  name: string
  users: number
  connections: number
  up: boolean
  [key: string]: unknown
}

defineProps<{instance: InstanceInfo}>()
</script>
