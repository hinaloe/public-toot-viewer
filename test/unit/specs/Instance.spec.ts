import InstanceListItem from '@/components/InstanceListItem.vue'
import {mount} from '@vue/test-utils'
import {describe, it, expect} from 'vitest'

describe('Instance list item', () => {
  it('should render Instance list item', () => {
    const instance = {
      "name": "mastodon.xyz",
      "uptime": 0.9322202579083936,
      "up": true,
      "https_score": null,
      "https_rank": null,
      "ipv6": false,
      "openRegistrations": true,
      "users": 23922,
      "statuses": "1651348",
      "connections": 18337,
      "info": {
          "optOut": false,
          "shortDescription": "Generalistic and moderated instance. All opinions are welcome, but hate speeches are prohibited.",
          "fullDescription": "Generalistic and moderated instance. All opinions are welcome, but hate speeches are prohibited.\r\n\r\nUsers who don't respect rules will be silenced, or suspended if they harass other people or if the the violation is considered too large.",
          "theme": null,
          "categories": [],
          "languages": [
              "en",
              "fr",
          ],
          "noOtherLanguages": false,
          "prohibitedContent": [
              "nudity_nocw",
              "pornography_nocw",
              "illegalContentLinks",
              "spam",
          ],
          "otherProhibitedContent": [],
      },
  }

  const wrapper = mount(InstanceListItem, {propsData: {instance}})

  expect(wrapper.html()).toMatchSnapshot()

  expect(wrapper.element.querySelector('.instance-name')?.textContent).to.equal('mastodon.xyz')
  expect(wrapper.html()).not.contains("DOWN")
  expect(wrapper.html()).includes('23922 users / connections: 18337')
  })

  it('should render Instance list item with DOWN status', () => {
    const instance = {
      "name": "physi.cc",
      "uptime": 0.9999999999896647,
      "up": false,
      "https_score": null,
      "https_rank": null,
      "ipv6": null,
      "openRegistrations": null,
      "users": null,
      "statuses": null,
      "connections": null,
  }

  const wrapper = mount(InstanceListItem, {propsData: {instance}})

  expect(wrapper.html()).toMatchSnapshot()

  expect(wrapper.element.querySelector('.instance-name')?.textContent).to.equal('physi.cc')
  expect(wrapper.html()).contains("DOWN")
  expect(wrapper.element.querySelector('b.text-danger')?.textContent?.trim()).to.equal('DOWN')
  })
})
