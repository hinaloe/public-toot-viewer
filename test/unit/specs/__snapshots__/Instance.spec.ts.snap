// Vitest Snapshot v1

exports[`Instance list item > should render Instance list item 1`] = `
"<a href=\\"#\\" class=\\"dropdown-item\\">
  <div class=\\"instance-name\\">mastodon.xyz</div>
  <div><small>
      <!----> 23922 users / connections: 18337 </small></div>
</a>"
`;

exports[`Instance list item > should render Instance list item with DOWN status 1`] = `
"<a href=\\"#\\" class=\\"dropdown-item\\">
  <div class=\\"instance-name\\">physi.cc</div>
  <div><small><b class=\\"text-danger\\">DOWN </b> users / connections: </small></div>
</a>"
`;
